core/src/test/java/org/opensearch/sql/expression/datetime/FromUnixTimeTest.java

```diff
@@ -0,0 +1,209 @@
+  @ParameterizedTest
+  @MethodSource("getDoubleSamples")
+  public void checkOfDouble(Double value) {
+    var intPart = Math.round(Math.floor(value));
+    var fracPart = value - intPart;
+    var valueAsString = new DecimalFormat("0.#").format(value);
+
+    assertEquals(
+        LocalDateTime.ofEpochSecond(intPart, (int)Math.round(fracPart * 1E9), ZoneOffset.UTC),
+        fromUnixTime(value),
+        valueAsString);
+    assertEquals(
+        LocalDateTime.ofEpochSecond(intPart, (int)Math.round(fracPart * 1E9), ZoneOffset.UTC),
+        eval(fromUnixTime(DSL.literal(new ExprDoubleValue(value)))).datetimeValue(),
+        valueAsString);
+  }
+
+  // We are testing all different formats and combinations, because they are already tested
+
+  private static Stream<Arguments> getLongSamplesWithFormat() {
+    return Stream.of(
+        Arguments.of(0L, "%c", "01"), // 1970-01-01 00:00:00, %c - 2 digit month
+        Arguments.of(1L, "%Y", "1970"), // 1970-01-01 00:00:01, %Y - 4 digit year
+        Arguments.of(1447430881L, "%s", "01"), // 2015-11-13 16:08:01, %s - second
+        Arguments.of(2147483647L, "%T", "03:14:07"), // 2038-01-19 03:14:07, %T - time
+        Arguments.of(1662577241L, "%d", "07") // 1662577241, %d - day of the month
+    );
+  }
+
+  /**
+   * Test processing different Long values with format.
+   * @param value a value
+   * @param format a format
+   * @param expected expected result
+   */
+  @ParameterizedTest
+  @MethodSource("getLongSamplesWithFormat")
+  public void checkOfLongWithFormat(Long value, String format, String expected) {
+    assertEquals(expected, fromUnixTime(value, format));
+    assertEquals(expected, eval(fromUnixTime(DSL.literal(new ExprLongValue(value)),
+        DSL.literal(new ExprStringValue(format)))).stringValue());
+  }
+
+  private static Stream<Arguments> getDoubleSamplesWithFormat() {
+    return Stream.of(
+        Arguments.of(0.123d, "%f", "123000"), // 1970-01-01 00:00:00.123, %f - microseconds
+        Arguments.of(100500.1005d, "%W", "Friday"), // 1970-01-02 03:55:00.1005, %W - Weekday name
+        Arguments.of(1447430881.56d, "%M", "November"), // 2015-11-13 16:08:01.56, %M - Month name
+        Arguments.of(2147483647.42d, "%j", "019"), // 2038-01-19 03:14:07.42, %j - day of the year
+        Arguments.of(1662577241.d, "%l", "7") // 2022-09-07 19:00:41, %l - 12 hour clock, no 0 pad
+    );
+  }
+
+  /**
+   * Test processing different Double values with format.
+   * @param value a value
+   * @param format a format
+   * @param expected expected result
+   */
+  @ParameterizedTest
+  @MethodSource("getDoubleSamplesWithFormat")
+  public void checkOfDoubleWithFormat(Double value, String format, String expected) {
+    assertEquals(expected, fromUnixTime(value, format));
+    assertEquals(expected, eval(fromUnixTime(DSL.literal(new ExprDoubleValue(value)),
+        DSL.literal(new ExprStringValue(format)))).stringValue());
+  }
+
+  @Test
+  public void checkInvalidValues() {
```

Comment:
These are worth splitting into separate test cases.

There's a lot of valuable information here  -- what input is invalid and how the function handles them, but at the moment one has to understand this code to learn that. 

A descriptive test name would make it easier to understand.

For example, 
- Is there a difference between
`assertEquals(ExprNullValue.of(), fromUnixTime(DSL.literal(-1L)).valueOf(null));` and `assertEquals(ExprNullValue.of(),
        fromUnixTime(DSL.literal(-1L), DSL.literal("%d")).valueOf(null));`? 
- Are `"%d"` and `"%q"` meaningful format strings or just arbitrary examples?

Reply:
> * Is there a difference between
>   `assertEquals(ExprNullValue.of(), fromUnixTime(DSL.literal(-1L)).valueOf(null));` and `assertEquals(ExprNullValue.of(), fromUnixTime(DSL.literal(-1L), DSL.literal("%d")).valueOf(null));`?

In both cases function should return `NULL`.

> * Are `"%d"` and `"%q"` meaningful format strings or just arbitrary examples?

`%d` => `dd` - Day of the month, numeric (`00..31`)
`%q` is invalid format option, interpreted as escaping for the next character (`q`).

Reply:
Updated in 541c63c2.